### DISCUSSION TITLE
Can I get the height during build?

### DISCUSSION THREAD

**Discussion by andersforsgren** [adamralph/minver#987] [Q&A] [answered]
I'd like to use a 4-component version number where 1.0.0-beta.123 gets file version 1.0.0.123,  so I'd like to access the git height. Is this possible?

**Comment by andersforsgren** [comment, marked as answer]
Solution I'm using at the moment: parsing it from the SemVer. This solves the problem.

```xml
<Height>$([System.Text.RegularExpressions.Regex]::Match($(MinVerPreRelease), `\w+\.(\d+)$`).Groups[1].Captures)</Height>
<FileVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).$(Height)</FileVersion>
```

**Comment by adamralph** [comment]
123 is not the git height in the example you've given. MinVer will only calculate that version from a tag, so you must have tagged the commit with `1.0.0-beta.123`.

I would advise against parsing the height. As stated in the [FAQ](https://github.com/adamralph/minver?tab=readme-ov-file#can-i-use-the-version-calculated-by-minver-for-other-purposes), the [official guidance for assembly file version](https://learn.microsoft.com/en-ca/dotnet/standard/library-guidance/versioning#assembly-file-version) states:

> ✔️ CONSIDER including a continuous integration build number as the AssemblyFileVersion revision.


The FAQ shows how to do that.
